### PR TITLE
File Structure

### PR DESCRIPTION
**Pull request by hassanelnajjar**
# File Structure
- [x] make the file structure as listed in #5 
- [x] install all the dependencies and devDependencies based in packages listed in #6
#5 #6

### CODE REVIEW

**Comment by Mu7ammadAbed** [comment]
I made some edits to the file structure especially with the `client/src` folder
check #5 please

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		"node": true,
+		"jest/globals": true
+	},
+	"extends": ["airbnb-base", "prettier", "plugin:jest/recommended"],
+	"globals": {
+		"Atomics": "readonly",
+		"SharedArrayBuffer": "readonly"
+	},
+	"parserOptions": {
+		"ecmaVersion": 2018
```
Comment (Mu7ammadAbed) [inline .eslintrc.json:15]:
```suggestion
		"ecmaVersion": 12
```

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+				"trailingComma": "all"
+			}
+		],
+		"jest/no-disabled-tests": "warn",
+		"jest/no-focused-tests": "error",
+		"jest/no-identical-title": "error",
+		"jest/prefer-to-have-length": "warn",
+		"jest/valid-expect": "error",
+		"consistent-return": "off",
+		"jest/no-test-callback": "off"
```
Comment (Mu7ammadAbed) [inline .eslintrc.json:32]: Remove all these rules

File: .eslintrc.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
+	"env": {
+		"browser": true,
+		"commonjs": true,
+		"es6": true,
```
Comment (Mu7ammadAbed) [inline .eslintrc.json:5]:
```suggestion
		"es2021": true,
```

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+	"editor.codeActionsOnSave": {
+      "source.fixAll": true,
+    },
+	"editor.formatOnSave": true,
+	"files.eol": "\n",
+	"cSpell.words": [
+		"pgport",
+		"usercalories",
+		"userexercisescalories",
+		"userfoodcalories"
```
Comment (Mu7ammadAbed) [inline .vscode/settings.json:22]: check these ;)

File: client/.eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
+{
+  "parser": "babel-eslint",
+  "extends": ["airbnb", "prettier","plugin:jest/recommended"],
+  "parserOptions": {
+    "es6": true,
+    "ecmaVersion": 6,
+    "ecmaFeatures": {
+      "impliedStrict": true,
```
Comment (Mu7ammadAbed) [inline client/.eslintrc.json:8]:
```suggestion
    "ecmaVersion": 12,
    "ecmaFeatures": {
      "jsx": true
      "impliedStrict": true,
```

File: client/.eslintrc.json [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      "node": {
+        "extensions": [".js", ".jsx"]
+      }
+    }
+  },
+  "root": true,
+  "env": {
+    "browser": true,
+    "jest/globals": true
+  },
```
Comment (Mu7ammadAbed) [inline client/.eslintrc.json:23]:
```suggestion
  "env": {
    "browser": true,
    "es2021": true,
    "jest/globals": true
  },
```

File: client/public/index.html [hunk 7]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+  <meta charset="utf-8" />
+  <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+  <meta name="viewport" content="width=device-width, initial-scale=1" />
+  <meta name="theme-color" content="#000000" />
+  <meta name="description" content="Web site created using create-react-app" />
+  <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
```
Comment (Mu7ammadAbed) [inline client/public/index.html:10]: Edit this line or remove it and remove the unnecessary icons.

File: client/src/App.test.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
```
Comment (Mu7ammadAbed) [inline client/src/App.test.js:1]: Remove this file

File: client/src/index.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+import App from './App';
+import reportWebVitals from './reportWebVitals';
```
Comment (Mu7ammadAbed) [inline client/src/index.js:5]: Remove this function and the `reportWebVitals` file 
```suggestion
```

File: client/src/reportWebVitals.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
+const reportWebVitals = (onPerfEntry) => {
```
Comment (Mu7ammadAbed) [inline client/src/reportWebVitals.js:1]: Remove this file

File: client/src/setupTests.js [hunk 11]
```diff
@@ -0,0 +1,5 @@
+// jest-dom adds custom jest matchers for asserting on DOM nodes.
```
Comment (Mu7ammadAbed) [inline client/src/setupTests.js:1]: Remove this file.

File: package.json [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+  "main": "index.js",
+  "scripts": {
+    "dev": "cross-env NODE_ENV=development nodemon server/index.js",
+    "kill": "fuser -k 5000/tcp",
+    "server": "npm run dev",
+    "client": "cd client && npm start",
+    "test": "cross-env NODE_ENV=test jest",
+    "kill-run": "npm run kill && npm run server",
+    "start": "cross-env NODE_ENV=production node ./server/index.js",
+    "heroku-postbuild": "NODE_ENV=production cd client && npm install && npm run build",
```
Comment (Mu7ammadAbed) [inline package.json:14]:
```suggestion
    "heroku-postbuild": "cd client && npm install && npm run build",
```

File: package.json [hunk 13]
```diff
@@ -0,0 +1,56 @@
+{
+  "name": "dentoro",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "dev": "cross-env NODE_ENV=development nodemon server/index.js",
+    "kill": "fuser -k 5000/tcp",
+    "server": "npm run dev",
```
Comment (Mu7ammadAbed) [inline package.json:9]:
```suggestion
```

File: package.json [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+    "client": "cd client && npm start",
+    "test": "cross-env NODE_ENV=test jest",
+    "kill-run": "npm run kill && npm run server",
+    "start": "cross-env NODE_ENV=production node ./server/index.js",
+    "heroku-postbuild": "NODE_ENV=production cd client && npm install && npm run build",
+    "build:db": "NODE_ENV=test node server/database/config",
```
Comment (Mu7ammadAbed) [inline package.json:15]:
```suggestion
    "build:db": "cross-env NODE_ENV=test node server/database/config",
```

File: package.json [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+  "dependencies": {
+    "bcrypt": "^5.0.1",
+    "compression": "^1.7.4",
+    "cookie": "^0.4.1",
+    "cookie-parser": "^1.4.5",
+    "cross-env": "^7.0.3",
+    "dotenv": "^8.2.0",
+    "express": "^4.17.1",
+    "jest": "^26.6.0",
+    "joi": "^17.4.0",
```
Comment (Mu7ammadAbed) [inline package.json:40]:
```suggestion
```

File: package.json [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/GSG-G9/dentoro/issues"
+  },
+  "homepage": "https://github.com/GSG-G9/dentoro#readme",
+  "dependencies": {
+    "bcrypt": "^5.0.1",
+    "compression": "^1.7.4",
+    "cookie": "^0.4.1",
```
Comment (Mu7ammadAbed) [inline package.json:34]:
```suggestion
```

**Review by Mu7ammadAbed — CHANGES_REQUESTED** [review]

File: .vscode/settings.json (hunk 4 above)
Comment (hassanelnajjar) [reply .vscode/settings.json:22]: removed :)

File: .eslintrc.json (hunk 3 above)
Comment (hassanelnajjar) [reply .eslintrc.json:5]: Done

File: .eslintrc.json (hunk 2 above)
Comment (hassanelnajjar) [reply .eslintrc.json:32]: removed :)

File: .eslintrc.json (hunk 1 above)
Comment (hassanelnajjar) [reply .eslintrc.json:15]: done

File: client/.eslintrc.json (hunk 5 above)
Comment (hassanelnajjar) [reply client/.eslintrc.json:8]: done

File: package.json (hunk 12 above)
Comment (hassanelnajjar) [reply package.json:14]: done

File: package.json (hunk 13 above)
Comment (hassanelnajjar) [reply package.json:9]: Did you want to remove dev and kill scripts?

File: package.json (hunk 14 above)
Comment (hassanelnajjar) [reply package.json:15]: done

File: package.json (hunk 16 above)
Comment (hassanelnajjar) [reply package.json:34]: done

**Comment by hassanelnajjar** [comment]
> I made some edits to the file structure especially with the `client/src` folder, and I transfer the `validation` folder into the `utils` folder,
> check #5, please

Done

File: client/.eslintrc.json (hunk 6 above)
Comment (hassanelnajjar) [reply client/.eslintrc.json:23]: done

File: client/public/index.html (hunk 7 above)
Comment (hassanelnajjar) [reply client/public/index.html:10]: done

File: client/src/index.js (hunk 9 above)
Comment (hassanelnajjar) [reply client/src/index.js:5]: done

File: client/src/reportWebVitals.js (hunk 10 above)
Comment (hassanelnajjar) [reply client/src/reportWebVitals.js:1]: done

**Review by Mu7ammadAbed — APPROVED** [review]

**Review by AlaaShurrab — APPROVED** [review]

**Review by Alaalser — APPROVED** [review]

**Pull request merged**